src/main/java/org/candlepin/subscriptions/tally/billing/admin/InternalBillingController.java

```diff
@@ -62,6 +69,37 @@ public List<MonthlyRemittance> process(BillableUsageRemittanceFilter filter) {
     return accountRemittanceList;
   }
 
+  public long processRetries(OffsetDateTime asOf) {
+    List<BillableUsageRemittanceEntity> remittances =
+        remittanceRepository.findByRetryAfterLessThan(asOf);
+    for (BillableUsageRemittanceEntity remittance : remittances) {
+      // re-trigger billable usage
+      billingProducer.produce(toBillableUsage(remittance));
+      // reset the retry after column
+      remittance.setRetryAfter(null);
+    }
+
+    // to save the retry after column for all the entities
+    remittanceRepository.saveAll(remittances);
+    return remittances.size();
+  }
+
+  private BillableUsage toBillableUsage(BillableUsageRemittanceEntity remittance) {
+    return new BillableUsage()
+        .withOrgId(remittance.getKey().getOrgId())
+        .withId(remittance.getTallyId())
+        .withSnapshotDate(remittance.getKey().getRemittancePendingDate())
+        .withProductId(remittance.getKey().getProductId())
+        .withSla(BillableUsage.Sla.fromValue(remittance.getKey().getSla()))
+        .withUsage(BillableUsage.Usage.fromValue(remittance.getKey().getUsage()))
+        .withBillingProvider(
+            BillableUsage.BillingProvider.fromValue(remittance.getKey().getBillingProvider()))
+        .withBillingAccountId(remittance.getKey().getBillingAccountId())
+        .withUom(remittance.getKey().getMetricId())
+        .withValue(remittance.getRemittedPendingValue())
```

Comment:
@kflahert is this the correct value to use?

Reply:
Yep, this looks good.